mri/operators/fourier/utils.py

```diff
@@ -255,37 +255,53 @@ def get_stacks_fourier(kspace_loc):
 
     Parameters
     ----------
-    ksapce_plane_loc: np.ndarray
+    kspace_plane_loc: np.ndarray
         the mask samples in the 3D Fourier domain.
-
+    shape: tuple
+        Reconstructed volume shape
     Returns
     ----------
-    ksapce_plane_loc: np.ndarray
+    kspace_plane_loc: np.ndarray
         A 2D array of samples which when stacked gives the 3D samples
     z_sample_loc: np.ndarray
         A 1D array of z-sample locations
     sort_pos: np.ndarray
         The sorting positions for opertor and inverse for incoming data
+    idx_mask_z: np.ndarray
+        contains the indices of the acquired Fourier planes
     """
     # Sort the incoming data based on Z, Y then X coordinates
     # This is done for easier stacking
     sort_pos = np.lexsort(tuple(kspace_loc[:, i]
                                 for i in np.arange(3)))
     kspace_loc = kspace_loc[sort_pos]
-    first_stack_len = np.size(np.where(
-        kspace_loc[:, 2] == np.min(kspace_loc[:, 2])))
+
+    # Find the mask used to sample stacks in z direction
+    full_stack_loc = convert_mask_to_locations(np.ones(shape[2]))[:, 0]
+    sampled_stack_loc = np.unique(kspace_loc[:, 2])
+
+    try:
+        idx_mask_z = np.asarray([np.where(x == full_stack_loc)[0][0] for
+                                 x in sampled_stack_loc])
+    except IndexError:
+        raise ValueError('The input must be a stack of 2D k-Space data')
+
+    first_stack_len = np.size(np.where(kspace_loc[:, 2] ==
+                                       np.min(kspace_loc[:, 2])))
     acq_num_slices = int(len(kspace_loc) / first_stack_len)
+
     stacked = np.reshape(kspace_loc, (acq_num_slices,
                                       first_stack_len, 3))
     z_expected_stacked = np.reshape(np.repeat(stacked[:, 0, 2],
                                               first_stack_len),
                                     (acq_num_slices,
                                      first_stack_len))
+
```

Comment:
Let's try to avoid as much as possible whiteline changes. They mess up the git history and clutter the PR.

Reply:
whiteline removed

Reply:
I don't see it removed